covalent/quantum/qserver/utils.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2023 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+import datetime
+import importlib
+from functools import lru_cache
+from typing import Any, Dict, List, Optional, Union
+
+import orjson
+from pydantic import BaseModel
+
+from ..._shared_files.qinfo import QNodeSpecs
+from ...executor.qbase import BaseQExecutor
+
+BATCH_ID_SEPARATOR = "@"
+MAX_DIFFERENT_EXECUTORS = 10
+
+
+class CircuitInfo(BaseModel):
```

Comment:
Hey @kessler-frost, we also need to have `Optional[]` here, no?

Seems like these fields usually get populated with valid values, but if you call a QElectron *without Covalent*, then `dispatch_id` and `electron_node_id` will remain `None`, which fails the type check versus `int` and `str` respectively.

There a number of fields apart from these that have default values `None`, without `Optional[]` wrapping the type hint.

Reply:
if it's not clear from the preview above, I'm referring to `CircuitInfo` specifically

Reply:
Yeah, I agree there were several fields that were `None` and should have the `Optional` type there. I initially only fixed the ones which were causing issues, but ideally yes, we should fix all of the "None" ones to include `Optional` in them.

Reply:
Do you mind if I push a commit to fix this?

Reply:
Oh please go ahead that'll be quite helpful 🙏